tests/MatchBot/Application/Matching/ArrayMatchingStorage.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace MatchBot\Tests\Application\Matching;
+
+use MatchBot\Application\RealTimeMatchingStorage;
+
+class ArrayMatchingStorage implements RealTimeMatchingStorage
+{
+    /** @var array<string,string> */
+    private array $storage;
+
+    private array $responses = [];
+
+    private bool $multiMode = false;
+
+    /**
+     * @var \Closure(string):void|null
+     */
+    private ?\Closure $preIncrCallback = null;
+
+    public function __construct()
+    {
+        $this->storage = [];
+    }
+
+    public function get(string $key): string|false
+    {
+        return $this->storage[$key] ?? false;
+    }
+
+    public function multi(): static
+    {
+        $this->multiMode = true;
+        return $this;
+    }
+
+    public function incrBy(string $key, int $increment): string|false|static
+    {
+        if ($this->preIncrCallback !== null) {
+            $callback = $this->preIncrCallback;
+            $callback($key);
+        }
+
+        $newValue = (float)($this->storage[$key] ?? 0) + $increment;
+        if (! $this->multiMode) {
+            return (string)$newValue;
+        }
+
+        $this->storage[$key] = (string) $newValue;
```

Comment:
```suggestion
        $this->storage[$key] = (string) $newValue;
        if (! $this->multiMode) {
            return (string)$newValue;
        }

```

Reply:
Thanks, done in 5f05d60 and updated affected test assertions.